client/packages/invoices/src/OutboundShipment/DetailView/AddFromMasterListButton.tsx

```diff
@@ -23,6 +28,8 @@ export const AddFromMasterListButtonComponent = () => {
           modalController.toggleOff();
           addFromMasterList(masterList);
         }}
+        filterBy={filterByName}
+        storeId={otherPartyStore?.id}
```

Comment:
I think storeId should still be logged in storeId ? (should we only add filterBy here ?). I think if you try this with otherParty that's a store but that's not active on this site (or for which user doesn't have permissions to log in, it might break)

Reply:
I think for request requisition filterBy should be storeId and for outbound shipment filterBy should be by otherPartyId (like it is here) ?

Reply:
I get the following (as per example where user doesn't have log in rights to a store for which outbound shipment is made):

![Jun-29-2022 16-37-22](https://user-images.githubusercontent.com/26194949/176352925-ac10c244-da38-485e-b514-38be7a4221e6.gif)

Reply:
> I think storeId should still be logged in storeId?

thanks - I think so too.. but if I do that then there are no master lists returned. There's a filterBy in the server which filters by the supplied `storeId` and if you also filter by name as well then you get nothing. The only way I've been able to return results is by using the store associated with the name.

I had tested using a store which isn't active on the current site and that's fine - still works.

The other option is to remove the restriction here: https://github.com/openmsupply/open-msupply/blob/develop/server/graphql/general/src/queries/master_list.rs#L115
and then remove the restriction here https://github.com/openmsupply/open-msupply/blob/develop/client/packages/system/src/MasterList/api/api.ts#L33 ( which I'd change to `filter: filterBy ?? { existsForStoreId: { equalTo: storeId } },`

Reply:
Yeah [this check ](https://github.com/openmsupply/open-msupply/blob/develop/server/graphql/general/src/queries/master_list.rs#L115)should not have been added, I think it should be removed.

I think f/e should be adding existsForStoreId (current logged in store) for request requisitions and eixstsForNameId for outbound shipment (and storeId in query for all end points being currently logged in store)